app/models/concerns/foreman_openscap/openscap_proxy_core_extensions.rb

```diff
@@ -4,18 +4,22 @@ module OpenscapProxyCoreExtensions
 
     included do
       validate :openscap_proxy_has_feature
+      validate :scap_client_class_present
+      after_save :update_scap_client
     end
 
-    def update_scap_client_params(proxy_id)
-      new_proxy = SmartProxy.find proxy_id
+    def update_scap_client
+      update_scap_client_params if openscap_proxy_id_changed? && openscap_proxy_id
```

Comment:
>I though lookup values are hidden, so what would be outdated?

Lookup values were the greatest pain, but there is still foreman_scap_client puppetclass. If we want to add/remove it together with openscap_proxy, it will not be displayed correctly. And we cannot hide it since we add classes with ajax, but remove them purely client-side.
 
If you just delete lookup values for host, the Puppet will IMHO use defaults (haven't tested). If you disassociate openscap_proxy, why would you send reports to Foreman? Granted, you can still have a host with assigned policy that has a cron, but removing a proxy is clearly a sign that you no longer care about reports from this host. If you still care about the host, you will remove the foreman_scap_client manually, or you will delete the host completely. In both cases: no more reports. Scenario when someone removes the proxy and keeps the foreman_scap_client does not make much sense to me. 

Setting lookup values to something that will cause errors when disassociating proxy is probably a good idea, that way we can catch when someone disassociated proxy "by accident".